doc-generator/src/main/scala/DocGenerator.scala

```diff
@@ -0,0 +1,95 @@
+import com.codacy.plugins.api._
+import com.codacy.plugins.api.results._
+import play.api.libs.json.Json
+import scala.xml._
+
+object DocGenerator extends App {
+  val scalastyleDefinition = XML.loadString(os.read(os.resource / "scalastyle_definition.xml"))
+  val scalastyleDocumentation = XML.loadString(os.read(os.resource / "scalastyle_documentation.xml"))
+
+  case class Checker(id: String, defaultLevel: String, cls: String, parameters: Seq[Checker.Parameter]) {
+    def patternId: String = cls.split('.').last
+
+    // Split camel case patternId into separate words
+    def title: String = patternId.stripSuffix("Checker").replaceAll("([a-z](?=[A-Z]))", "$1 ")
+
+    def category: Pattern.Category = patternId match {
+      case "VarFieldChecker" | "VarLocalChecker" | "ProcedureDeclarationChecker" | "NotImplementedErrorUsage" |
+          "EqualsHashCodeChecker" | "NullChecker" | "NoCloneChecker" | "NoFinalizeChecker" | "CovariantEqualsChecker" |
+          "StructuralTypeChecker" | "PublicMethodsHaveTypeChecker" =>
+        Pattern.Category.ErrorProne
+      case _ => Pattern.Category.CodeStyle
+    }
+
+    def level: Result.Level = defaultLevel match {
+      case "warning" => Result.Level.Warn
+      case "error" => Result.Level.Err
+      case _ => Result.Level.Info
+    }
+  }
+
+  object Checker {
+    case class Parameter(name: String, default: String)
+  }
+
+  val checkers = scalastyleDefinition
+    .\("checker")
+    .map(
+      n =>
```

Comment:
what does "n" stand for?

Reply:
It stands for `badNaming`..
Changing it 👌

Reply:
Now I remember, `n` was for "node" because it's an xml node. Renamed now to `node`